crates/stark/src/prover.rs

```diff
@@ -25,8 +26,10 @@ use crate::{
 };
 
 /// An algorithmic & hardware independent prover implementation for any [`MachineAir`].
-pub trait MachineProver<SC: StarkGenericConfig, A: MachineAir<SC::Val>>:
-    'static + Send + Sync
+pub trait MachineProver<
+    SC: StarkGenericConfig,
+    A: MachineAir<SC::Val> + Air<SymbolicAirBuilder<SC::Val>>,
```

Comment:
It would be great if `Air<SymbolicAirBuilder<SC::Val>>,` trait bound can be removed here, esp. since `MachineProver` is one of the main traits throughout the whole `sp1` repo.